packages/js/product-editor/src/hooks/use-product.ts

```diff
@@ -0,0 +1,46 @@
+/**
+ * External dependencies
+ */
+import { applyFilters } from '@wordpress/hooks';
+import { Product, ReadOnlyProperties } from '@woocommerce/data';
+// eslint-disable-next-line @typescript-eslint/ban-ts-comment
+// @ts-ignore No types for this exist yet.
+// eslint-disable-next-line @woocommerce/dependency-group
+import { useEntityId } from '@wordpress/core-data';
+import { useDispatch } from '@wordpress/data';
+
+
+export function useProduct() {
+	const { editEntityRecord, saveEntityRecord } = useDispatch( 'core' );
+
+	const productId = useEntityId( 'postType', 'product' );
+
+    function editProduct( data: Omit< Product, ReadOnlyProperties > ) {
+        return editEntityRecord(
+            'postType',
+            'product',
+            productId,
+            applyFilters(
+                'woocommerce_edit_product_data',
```

Comment:
We discussed the possibility of using a naming convention similar to "core". The following filters are available:

```
'blockEditor.__unstableCanInsertBlockType'
'blockEditor.useSetting.before'
'blockEditor.wrapSelectionSettings'
'blocks.getBlockAttributes'
'blocks.getBlockDefaultClassName'
'blocks.getBlockMenuDefaultClassName'
'blocks.getSaveContent.extraProps'
'blocks.getSaveElement'
'blocks.navigation.__unstableMenuItemsToBlocks'
'blocks.registerBlockType'
'blocks.switchToBlockType.transformedBlock'
'editor.Autocomplete.completers'
'editor.PostFeaturedImage.imageSize'
'editor.PostPreview.interstitialMarkup'
'editor.__unstableSavePost'
'media.crossOrigin'
'native.block_editor_props'
'native.missing_block_detail'
'native.persist-html'
'native.supported_endpoints'
'native.unsupported_block_details_extra_note'
'native.unsupported_block_details_web_editor_action'
'plugins.registerPlugin'
'transform'
```

`woo.product.data.edit` and `woo.product.data.save`? Too many dots actually